PhysicsTools/NanoAOD/python/nano_cff.py

```diff
@@ -138,16 +138,30 @@ def nanoAOD_addBoostedTauIds(process, idsToRun=[]):
 
     return process
 
-def nanoAOD_addPNetToTaus(process, addPNetInfo=False, runPNetCHSAK4=False):
-    if addPNetInfo:
+def nanoAOD_addUTagToTaus(process, addUTagInfo=False, runUTagAK4=False,
+                          usePUPPIjets=False):
+    if addUTagInfo:
         originalTauName = process.finalTaus.src.value()
-        updatedTauName = originalTauName+'WithPNet'
-        jetCollection = "updatedJets"
-        process.load('RecoBTag.ONNXRuntime.pfParticleNetFromMiniAODAK4_cff')
-        pnetTagName = "pfParticleNetFromMiniAODAK4CHSCentralJetTag"
-        pnetDiscriminators = [];
-        for tag in getattr(process,pnetTagName+"s").flav_names.value():
-            pnetDiscriminators.append(pnetTagName+"s:"+tag)
+        
+        if usePUPPIjets: # option to use PUPPI jets   
+            jetCollection = "updatedJetsPuppi"
+            TagName = "pfUnifiedParticleTransformerAK4JetTag"
+            tag_prefix = "byUTagPUPPI"
+            updatedTauName = originalTauName+'WithUTagPUPPI'
+            # Unified ParT Tagger used for PUPPI jets
+            from RecoBTag.ONNXRuntime.pfUnifiedParticleTransformerAK4JetTags_cfi import pfUnifiedParticleTransformerAK4JetTags
+            process.pfUnifiedParticleTransformerAK4JetTags = pfUnifiedParticleTransformerAK4JetTags.clone()
+        else: # use CHS jets by default
+            jetCollection = "updatedJets"
+            TagName = "pfParticleNetFromMiniAODAK4CHSCentralJetTag"
+            tag_prefix = "byUTagCHS"
+            updatedTauName = originalTauName+'WithUTagCHS'
+            # PNet tagger used for CHS jets
+            process.load('RecoBTag.ONNXRuntime.pfParticleNetFromMiniAODAK4_cff')
```

Comment:
```suggestion
            from RecoBTag.ONNXRuntime.pfParticleNetFromMiniAODAK4_cff import pfParticleNetFromMiniAODAK4CHSCentralJetTags
            Discriminators = pfParticleNetFromMiniAODAK4CHSCentralJetTags.flav_names.value()
```

Reply:
Thanks, will implement

Reply:
In the end for this, we didn't go for this exact form (we decided to add the tag name in front of the discriminator names for a safety/cross check in our producer) - but I used imports instead of adding to process as requested